### ISSUE TITLE
make lint fails multiple checks

### ISSUE THREAD

**Issue by suo** [pytorch/pytorch#56738]
I ran `make setup_lint` then `make lint`. Ran in to the following problems:
1. Looks like we're not pinning the mypy version?
```
❌ mypy: Run mypy
You are using mypy version 0.770, which is not supported
in the PyTorch repo. Please switch to mypy version 0.812.

For example, if you installed mypy via pip, run this:

    pip install mypy==0.812

Or if you installed mypy via conda, run this:

    conda install -c conda-forge mypy=0.812
tools/print_test_stats.py:18:1: error: No library stub file for standard library module 'xml.dom'  [import]
tools/print_test_stats.py:18:1: note: (Stub files are from https://github.com/python/typeshed)
Found 1 error in 1 file (checked 77 source files)
```

2. false positives for ensure no tabs?
```
❌ quick-checks: Ensure no tabs
.circleci/config.yml:1260:          export COMMAND='((echo "sudo chown -R jenkins workspace && cd workspace && '"export CIRCLE_SHA1='$CIRCLE_SHA1'"' && . ./.circleci/scripts/python_doc_push_script.sh docs/'$target' '$target' site") | docker exec -u jenkins -i "$id" bash) 2>&1'
.circleci/config.yml:1305:          export COMMAND='((echo "sudo chown -R jenkins workspace && cd workspace && '"export CIRCLE_SHA1='$CIRCLE_SHA1'"' && . ./.circleci/scripts/cpp_doc_push_script.sh docs/"$target" master") | docker exec -u jenkins -i "$id" bash) 2>&1'
.circleci/docker/build.sh:331:       -t "$tmp_tag" \
.circleci/docker/build.sh:344:  docker run --rm "$tmp_tag" $*
.circleci/verbatim-sources/job-specs/job-specs-custom.yml:52:          export COMMAND='((echo "sudo chown -R jenkins workspace && cd workspace && '"export CIRCLE_SHA1='$CIRCLE_SHA1'"' && . ./.circleci/scripts/python_doc_push_script.sh docs/'$target' '$target' site") | docker exec -u jenkins -i "$id" bash) 2>&1'
.circleci/verbatim-sources/job-specs/job-specs-custom.yml:97:          export COMMAND='((echo "sudo chown -R jenkins workspace && cd workspace && '"export CIRCLE_SHA1='$CIRCLE_SHA1'"' && . ./.circleci/scripts/cpp_doc_push_script.sh docs/"$target" master") | docker exec -u jenkins -i "$id" bash) 2>&1'
.jenkins/caffe2/test.sh:47:    case "$test" in
.jenkins/caffe2/test.sh:56:          LD_LIBRARY_PATH="$ld_library_path" "$test"
.jenkins/caffe2/test.sh:60:        LD_LIBRARY_PATH="$ld_library_path" "$test" --benchmark_color=false
.jenkins/caffe2/test.sh:74:            "$test" --gtest_output=xml:"$gtest_reports_dir/$(basename $test).xml"
caffe2/operators/cross_entropy_op.cc:379:where ($i$, $j$) is the classifier's prediction of the $j$th class (the correct one), and $i$ is the batch size. Each log has a lower limit for numerical stability.
caffe2/operators/cross_entropy_op.cc:381:The difference between *LabelCrossEntropy* and *CrossEntropy* is how the labels are specified. Here, the labels are a length $N$ list of integers, whereas in CrossEntropy the labels are a $NxD$ dimensional matrix of one hot label vectors. However, the results of computation should be the same, as shown in the two examples where ($i$, $j$) is the classifier's prediction of the $j$th class (the correct one), and $i$ is the batch size. Each log has a lower limit for numerical stability.
caffe2/operators/cross_entropy_op.cc:448:        "Blob containing the labels used to compare the input. $label$ is a length $N$ list of integers, where each element is the integer label for the $n$th element of the batch.")
caffe2/operators/cross_entropy_op.cc:612:where ($i$, $j$) is the classifier's prediction of the $j$th class (the correct one), and $i$ is the batch size. Each log has a lower limit for numerical stability.
caffe2/operators/is_empty_op.cc:10:The *IsEmpty* op accepts a single input $tensor$, and produces a single boolean output $is\_empty$. The output is *True* if and only if $tensor$ has size == 0.
docs/caffe2/.Doxyfile-c:768:# can produce. The string should contain the $file, $line, and $text tags, which
docs/caffe2/.Doxyfile-c:773:# The default value is: $file:$line: $text.
docs/caffe2/.Doxyfile-c:775:WARN_FORMAT            = "$file:$line: $text"
docs/caffe2/.Doxyfile-c:1710:# following commands have a special meaning inside the header: $title,
docs/caffe2/.Doxyfile-c:1712:# $projectbrief, $projectlogo. Doxygen will replace $title with the empty
docs/caffe2/.Doxyfile-python:768:# can produce. The string should contain the $file, $line, and $text tags, which
docs/caffe2/.Doxyfile-python:773:# The default value is: $file:$line: $text.
docs/caffe2/.Doxyfile-python:775:WARN_FORMAT            = "$file:$line: $text"
docs/caffe2/.Doxyfile-python:1708:# following commands have a special meaning inside the header: $title,
docs/caffe2/.Doxyfile-python:1710:# $projectbrief, $projectlogo. Doxygen will replace $title with the empty
docs/caffe2/header.html:9:<!--BEGIN PROJECT_NAME--><title>$projectname: $title</title><!--END PROJECT_NAME-->
docs/caffe2/header.html:10:<!--BEGIN !PROJECT_NAME--><title>$title</title><!--END !PROJECT_NAME-->
docs/caffe2/header.html:15:$treeview
ios/TestApp/bootstrap.sh:43:        reply=$(bash -c 'read -r -p "Team Id:" tmp; echo $tmp')
ios/TestApp/bootstrap.sh:48:        reply=$(bash -c 'read -r -p "Provisioning Profile:" tmp; echo $tmp')
scripts/fbcode-dev-setup/onnx_c2_setup.sh:68:  mv --backup=t -T "$onnx_root" "${onnx_root}.old.$timestamp"
scripts/fbcode-dev-setup/onnx_c2_setup.sh:73:  mv --backup=t -T "$onnx_root_link" "${onnx_root_link}.old.$timestamp"
scripts/onnx/install-develop.sh:8:tp2_dir="$top_dir/third_party"
scripts/onnx/install-develop.sh:13:pip install --no-use-pep517 -e "$tp2_dir/onnx"
scripts/onnx/install-develop.sh:16:pip install -r "$top_dir/caffe2/requirements.txt"
scripts/onnx/install-develop.sh:17:pip install -r "$top_dir/requirements.txt"
scripts/onnx/install.sh:8:tp2_dir="$top_dir/third_party"
scripts/onnx/install.sh:9:BUILD_DIR="$top_dir/build"
scripts/onnx/install.sh:33:_pip_install -b "$BUILD_DIR/onnx" "file://$tp2_dir/onnx#egg=onnx"
scripts/onnx/install.sh:36:pip install -r "$top_dir/caffe2/requirements.txt"
scripts/onnx/install.sh:37:pip install -r "$top_dir/requirements.txt"
scripts/onnx/test.sh:37:    "$top_dir/test/onnx"
scripts/onnx/test.sh:55:  --ignore "$top_dir/test/onnx/test_pytorch_onnx_onnxruntime.py" \
scripts/onnx/test.sh:56:  --ignore "$top_dir/test/onnx/test_custom_ops.py" \
scripts/onnx/test.sh:57:  --ignore "$top_dir/test/onnx/test_models_onnxruntime.py" \
scripts/onnx/test.sh:58:  --ignore "$top_dir/test/onnx/test_utility_funs.py" \
scripts/onnx/test.sh:59:  --ignore "$top_dir/test/onnx/test_pytorch_onnx_caffe2.py" \
scripts/onnx/test.sh:60:  --ignore "$top_dir/test/onnx/test_pytorch_onnx_shape_inference.py" \
scripts/onnx/test.sh:61:  --ignore "$top_dir/test/onnx/test_pytorch_onnx_onnxruntime_cuda.py" \
scripts/onnx/test.sh:62:  --ignore "$top_dir/test/onnx/test_pytorch_onnx_caffe2_quantized.py" \
scripts/onnx/test.sh:69:    "$top_dir/test/onnx/test_pytorch_onnx_onnxruntime.py::TestONNXRuntime_opset7" \
scripts/onnx/test.sh:70:    "$top_dir/test/onnx/test_pytorch_onnx_onnxruntime.py::TestONNXRuntime_opset8" \
scripts/onnx/test.sh:71:    "$top_dir/test/onnx/test_pytorch_onnx_onnxruntime.py::TestONNXRuntime" \
scripts/onnx/test.sh:72:    "$top_dir/test/onnx/test_custom_ops.py" \
scripts/onnx/test.sh:73:    "$top_dir/test/onnx/test_models_onnxruntime.py" \
scripts/onnx/test.sh:74:    "$top_dir/test/onnx/test_utility_funs.py" \
scripts/onnx/test.sh:75:    "$top_dir/test/onnx/test_pytorch_onnx_caffe2.py" \
scripts/onnx/test.sh:76:    "$top_dir/test/onnx/test_pytorch_onnx_caffe2_quantized.py"
scripts/onnx/test.sh:82:      "$top_dir/test/onnx/test_pytorch_onnx_onnxruntime.py::TestONNXRuntime_opset$i"
torch/csrc/jit/codegen/cuda/docs/fuser.doxygen:768:# can produce. The string should contain the $file, $line, and $text tags, which
torch/csrc/jit/codegen/cuda/docs/fuser.doxygen:773:# The default value is: $file:$line: $text.
torch/csrc/jit/codegen/cuda/docs/fuser.doxygen:775:WARN_FORMAT            = "$file:$line: $text"
torch/csrc/jit/codegen/cuda/docs/fuser.doxygen:1756:# following commands have a special meaning inside the header: $title,
torch/csrc/jit/codegen/cuda/docs/fuser.doxygen:1758:# $projectbrief, $projectlogo. Doxygen will replace $title with the empty
torch/csrc/jit/frontend/ir_emitter.cpp:2505:    const auto tmp_name = createTempName("$tmp_assign_");
The above lines have tabs; please convert them to spaces
```

3. setup_lint does not install shellcheck
```
make shellcheck-gha
make[1]: Entering directory '/raid/suo/pytorch'
tools/extract_scripts.py --out=.shellcheck_generated_gha
tools/run_shellcheck.sh .shellcheck_generated_gha
xargs: shellcheck: No such file or directory
make[1]: *** [Makefile:28: shellcheck-gha] Error 127
make[1]: Leaving directory '/raid/suo/pytorch'
make: *** [Makefile:32: generate-gha-workflows] Error 2
```

**Comment by driazati** [comment]
What OS/git version are you on?

```bash
cat /etc/os-release
git --version
```

**Comment by suo** [comment]
```
NAME="Ubuntu"
VERSION="20.04.1 LTS (Focal Fossa)"
ID=ubuntu
ID_LIKE=debian
PRETTY_NAME="Ubuntu 20.04.1 LTS"
VERSION_ID="20.04"
HOME_URL="https://www.ubuntu.com/"
SUPPORT_URL="https://help.ubuntu.com/"
BUG_REPORT_URL="https://bugs.launchpad.net/ubuntu/"
PRIVACY_POLICY_URL="https://www.ubuntu.com/legal/terms-and-policies/privacy-policy"
VERSION_CODENAME=focal
UBUNTU_CODENAME=focal
git version 2.31.1
```

After running `pip install mypy==0.812` and running `make check`, mypy job fails with:
```
❌ mypy: Run mypy
Success: no issues found in 77 source files
torch/onnx/utils.py:204: error: Too many arguments for "_jit_pass_onnx_scalar_type_analysis"  [call-arg]
torch/package/analyze/debug_pickler.py:76: error: Name '_format_obj_stack' is not defined  [name-defined]
Found 2 errors in 2 files (checked 1327 source files)
```

**Comment by suo** [comment]
Also, the `flake8-py3` job appears to succeed, but there is a noted issue:
```
✅ flake8-py3: Run flake8
./torch/package/analyze/debug_pickler.py:8:1: B903 Data class should either be immutable or use __slots__ to save memory. Use collections.namedtuple to generate an immutable class, or enumerate the attributes in a __slot__ declaration in the class to leave attributes mutable.
```

and when I run `flake8` without any arguments, it fails.

(welcome back btw :P )

**Comment by driazati** [comment]
> `./torch/package/analyze/debug_pickler.py` is a file in my local checkout, I can push a branch if that is helpful.

A branch would be great, the bugs in the main issue are fixed in #56752
